sys/net/netdev_test/ieee802154/netdev_test_ieee802154.c

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (C) 2016-21 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @{
+ *
+ * @file
+ * @author  Martine S. Lenders <m.lenders@fu-berlin.de>
+ */
+
+#include <string.h>
+
+#include "mutex.h"
+
+#include "net/netdev_test/ieee802154.h"
+
+static int _get_device_type(netdev_t *dev, void *value, size_t max_len)
+{
+    (void)dev;
+    assert(max_len == sizeof(uint16_t));
+    *((uint16_t *)value) = NETDEV_TYPE_IEEE802154;
+    return sizeof(uint16_t);
+}
+
+static int _get_max_packet_size(netdev_t *dev, void *value, size_t max_len)
+{
+    (void)dev;
+    assert(max_len == sizeof(uint16_t));
+    *((uint16_t *)value) = 102U;
+    return sizeof(uint16_t);
+}
+
+static int _get_src_len(netdev_t *dev, void *value, size_t max_len)
+{
+    (void)dev;
+    assert(max_len == sizeof(uint16_t));
+    *((uint16_t *)value) = IEEE802154_LONG_ADDRESS_LEN;
+    return sizeof(uint16_t);
+}
+
+static int _get_address_long(netdev_t *dev, void *value, size_t max_len)
+{
+    netdev_test_ieee802154_t *netdev = (netdev_test_ieee802154_t *)dev;
+    assert(max_len >= sizeof(netdev->eui64));
+    memcpy(value, &netdev->eui64, sizeof(netdev->eui64));
+    return sizeof(netdev->eui64);
+}
+
+static int _get_proto(netdev_t *dev, void *value, size_t max_len)
+{
+    (void)dev;
+#if IS_USED(MODULE_GNRC_NETTYPE_SIXLOWPAN)
+    assert(max_len == sizeof(gnrc_nettype_t));
+    *((gnrc_nettype_t *)value) = GNRC_NETTYPE_SIXLOWPAN;
+    return sizeof(gnrc_nettype_t);
+#else
+    (void)value;
+    (void)max_len;
+    return -ENOTSUP;
+#endif
+}
+
+void netdev_test_ieee802154_setup(netdev_test_ieee802154_t *netdev, void *state,
+                                  const eui64_t *eui64)
+{
+    netdev_test_setup(&netdev->netdev, state);
+    mutex_lock(&netdev->netdev.mutex);
+    if (eui64 != NULL) {
+        memcpy(&netdev->eui64, eui64, sizeof(netdev->eui64));
+    }
+    else {
+        netdev_eui64_get((netdev_t *)netdev, &netdev->eui64);
```

Comment:
```suggestion
        netdev_eui64_get(&netdev->netdev, &netdev->eui64);
```

I think with something like this the explicit cast can be avoided here.

Reply:
It goes much deeper since `netdev_test_ieee802154_t` is a subclass of `netdev_test_t` which is a subclass of `netdev_ieee802154_t` which is a subclass of `netdev_t` (see https://github.com/RIOT-OS/RIOT/pull/16020/commits/456c2095b8a824bcd33c8d56b79ffeaa9e27c1bc). If that is too deep and we decide to rather go for the cast, we can still revert https://github.com/RIOT-OS/RIOT/pull/16020/commits/456c2095b8a824bcd33c8d56b79ffeaa9e27c1bc.